internal/query/user_membership.go

```diff
@@ -184,18 +183,42 @@ var (
 		table: membershipAlias,
 	}
 
-	membershipFrom = "(" +
-		prepareOrgMember() +
-		" UNION ALL " +
-		prepareIAMMember() +
-		" UNION ALL " +
-		prepareProjectMember() +
-		" UNION ALL " +
-		prepareProjectGrantMember() +
-		") AS " + membershipAlias.identifier()
+	membershipOwnerRemoved = Column{
+		name:  projection.MemberOwnerRemoved,
+		table: membershipAlias,
+	}
+	membershipOwnerRemovedUser = Column{
+		name:  projection.MemberUserOwnerRemoved,
+		table: membershipAlias,
+	}
+	membershipGrantedOrgRemoved = Column{
+		name:  projection.ProjectGrantMemberGrantedOrgRemoved,
+		table: membershipAlias,
+	}
 )
 
-func prepareMembershipsQuery() (sq.SelectBuilder, func(*sql.Rows) (*Memberships, error)) {
+func getMembershipFromQuery(withOwnerRemoved bool) (string, []interface{}) {
```

Comment:
we've removed params from the prepare functions and state all conditions directly in the "where" for better reading